decidim-core/lib/decidim/scopable_component.rb

```diff
@@ -16,20 +16,30 @@ module ScopableComponent
       #
       # Returns a boolean.
       def has_subscopes?
-        (scopes_enabled? || participatory_space.scopes_enabled?) && subscopes.any?
+        (scopes_enabled? || parent_scopes_enabled?) && subscopes.any?
       end
 
       # Public: Returns the component Scope
       def scope
-        return participatory_space.scope unless scopes_enabled?
+        return parent_component_or_space.scope unless scopes_enabled?
 
-        participatory_space.scopes.find_by(id: settings.scope_id)
+        parent_component_or_space.scopes.find_by(id: settings.scope_id)
       end
 
       # Returns a boolean.
       def scopes_enabled
         settings.try(:scopes_enabled)
       end
+
+      def parent_component_or_space
+        return parent if parent
+
+        participatory_space
+      end
+
+      def parent_scopes_enabled?
```

Comment:
do you think that it would be simpler as a delegated method? (It would change the name too, but I think that it would be more accurate too 😬)

`delegate :scopes_enabled, to: :parent_component_or_space, prefix: true`

Reply:
Not sure about it, because this method should return `true` if **any** of the ancestors have scopes enabled. 

So this method should return `true` when has a `parent` (component) with scopes disabled, but  the `participatory_space` has scopes enabled; If it was delegated to `parent_component_or_space`, I understand that it would return a `false`.

Maybe I could rename the method to help better understand the functionality, `an_ancestor_has_scopes_enabled?`.

This is used in `:has_subscopes?` (line 18) of this file.

Reply:
ah, ok, then I like your renaming proposal. :+1: